### DISCUSSION TITLE
Improve Performance of Tooling and Other Calls

### DISCUSSION THREAD

**Discussion by JosephAllen** [forcedotcom/salesforcedx-vscode#5777] [Ideas]
If we use `const sfdxCoreExports = vscode.extensions.getExtension('salesforce.salesforcedx-vscode-core')!.exports;`, we may be able to remove the need for the `@salesforce\core` dependecy.

I created a simple Extension that executes anonymous apex. I am seeing 3-10 times better performance using an `extensionDependencies` to `salesforce.salesforcedx-vscode-core` and implementing `vscode.extensions.getExtension`

Here is the extension I created: [apex-code-runner](https://github.com/JosephAllen/apex-code-runner)

Even when I call the auth methods before my callouts to the SOAP Apex api, I am still seeing much better performance that the Extension Pack command that do the same.

Let me know how I can help.

Certified Application Arch, Certified Systems Arch, Dev II Certification

**Comment by lcampos** [comment]
Hi @JosephAllen, thanks for opening this and providing feedback. We're currently focusing on performance work for deploys & retrieves [Performance Enhancements](https://salesforcedx-vscode.netlify.app/tools/vscode/en/user-guide/perf-enhancements) and will soon start working on the Apex commands (running tests, anon execution, etc). The approach we'll follow is to run this from a new library which is a bit faster than running this from the core extension exports and current implementation.

I'll make sure to update this thread once we start publishing the changes.
cc @smaddox-sf

**Comment by JosephAllen** [comment]
@lcampos  , thanks. From what I am seeing the issue appears to be authenticating and then using parts of the cli, versus the SOAP API. With a valid token, anon apex runs really fast with a direct SOAP call. That's what my stop gap extension  does.
